Microsoft.Azure.Cosmos/src/Serializer/CosmosSystemTextJsonSerializer.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
+{
+    using System;
+    using System.IO;
+    using System.Reflection;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    /// <summary>
+    /// This class provides a default implementation of System.Text.Json Cosmos Linq Serializer.
+    /// </summary>
+    public class CosmosSystemTextJsonSerializer : CosmosLinqSerializer
+    {
+        /// <summary>
+        /// A read-only instance of <see cref="JsonSerializerOptions"/>.
+        /// </summary>
+        private readonly JsonSerializerOptions jsonSerializerOptions;
+
+        /// <summary>
+        /// Creates an instance of <see cref="CosmosSystemTextJsonSerializer"/>
+        /// with the default values for the Cosmos SDK
+        /// </summary>
+        /// <param name="jsonSerializerOptions">An instance of <see cref="JsonSerializerOptions"/> containing the json serialization options.</param>
+        public CosmosSystemTextJsonSerializer(
+            JsonSerializerOptions jsonSerializerOptions)
+        {
+            this.jsonSerializerOptions = jsonSerializerOptions;
+        }
+
+        /// <inheritdoc/>
+        public override T FromStream<T>(Stream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
+
+            if (stream.CanSeek && stream.Length == 0)
+            {
+                return default;
+            }
+
+            if (typeof(Stream).IsAssignableFrom(typeof(T)))
+            {
+                return (T)(object)stream;
+            }
+
+            using (stream)
+            {
+                using StreamReader reader = new (stream);
+                return JsonSerializer.Deserialize<T>(reader.ReadToEnd(), this.jsonSerializerOptions);
```

Comment:
```suggestion
                return JsonSerializer.Deserialize<T>(stream, this.jsonSerializerOptions);
```
STJ is able to deserialize the stream directly without needing to allocate an intermediate string.

Reply:
Right now, the `System.Text.JSON` package depends on the `Azure.Core` version `1.19.0`, which uses the STJ version `4.6.0`. This version of the STJ serializer doesn't support the `Deserialize()` directly with a `stream`. Hence the conversion is needed at the moment.

Reply:
Ok, clear. Can you add a TODO not to forget to update the implementation once the version of STJ will be updated?

Reply:
snap! https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4332#discussion_r1520153811